broadcast/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+//! Replication of messages across a network.
+
+use bytes::Bytes;
+use commonware_cryptography::Array;
+use futures::channel::oneshot;
+use std::future::Future;
+
+pub mod linked;
+
+/// Proof is a blob that attests to some data.
+pub type Proof = Bytes;
+
+/// Broadcaster is the interface responsible for replication of messages across a network.
+pub trait Broadcaster: Clone + Send + 'static {
+    /// Digest is an arbitrary hash digest.
+    type Digest: Array;
+
+    /// Attempt to broadcast a digest to the network.
+    ///
+    /// Returns a future that resolves to a boolean indicating success.
```

Comment:
nit: I would add a little more detail to "why failure?" -> more than just networking, it could build off wrong parent, etc.

If you don't think there is a good way to do this without being too "coupled" with linked, disregard.

Reply:
The issue currently seems to be that this is a bit of a poor interface. There isn't:
- a clear failure code
- a function to query whether the broadcaster is ready to broadcast a new chunk

I don't know whether either of these is "more correct" but it's clear the current interface is lacking a bit. That said, I'll update the comment very slightly